src/blueprints/settings/__init__.py

```
@@ -352,24 +372,97 @@ def _set_update_state(running: bool, unit: str | None):
+
+
+def _validate_update_target_tag(target_tag: str | None) -> str | None:
+    """Return a sanitised semver target tag or ``None``; raise on bad input.
+
+    CodeQL recognises a successful ``re.fullmatch`` *guard* against a fixed
+    safe pattern as a sanitiser for ``py/command-line-injection``.
+    """
+    if target_tag is None:
+        return None
+    if not isinstance(target_tag, str):
+        raise ValueError(f"Invalid target tag format: {target_tag!r}")
```

Ruff (0.15.9)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)